File: backend/CarMaintenanceTracker.Api/Entities/MaintenanceTemplate.cs
using System.ComponentModel.DataAnnotations;

namespace CarMaintenanceTracker.Api.Entities;

public class MaintenanceTemplate
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(10)]
    public string IntervalType { get; set; } = string.Empty;

    [Required]
    public int IntervalValue { get; set; }

    public bool Archived { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/maintenance-template.ts
export interface MaintenanceTemplate {
  id: number;
  name: string;
  intervalType: 'km' | 'time';
  intervalValue: number;
  archived: boolean;
  createdAt: string;
}

export interface CreateMaintenanceTemplateDto {
  name: string;
  intervalType: string;
  intervalValue: number;
}

export interface ValidationError {
  errors: {
    [key: string]: string[];
  };
}
